File: CS-Examples/24_Workbook/DecryptWorkbook.vb
Imports Spire.Xls
Imports System.ComponentModel
Imports System.Text

Namespace DecryptWorkbook
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Specify the file name of the workbook to decrypt
            Dim fileName As String = "..\..\..\..\..\..\Data\DecryptWorkbook.xlsx"

            ' Check if the workbook is password-protected
            Dim value As Boolean = Workbook.IsPasswordProtected(fileName)

            ' If the workbook is password-protected, perform the following operations
            If value Then

                ' Create a new Workbook object
                Dim workbook As New Workbook()

                ' Set the open password for the workbook
                workbook.OpenPassword = "eiceblue"

                ' Load the password-protected workbook
                workbook.LoadFromFile(fileName)

                ' Remove the protection from the workbook
                workbook.UnProtect()

                ' Save the decrypted workbook to a new file with Excel Version 2010 format
                workbook.SaveToFile("DecryptWorkbook_result.xlsx", ExcelVersion.Version2010)

                ' Release the resources used by the workbook
                workbook.Dispose()
            End If

            ExcelDocViewer("DecryptWorkbook_result.xlsx")
		End Sub
		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace

File: CS-Examples/24_Workbook/DecryptWorkbook.cs
using Spire.Xls;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace DecryptWorkbook
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
        private void btnRun_Click(object sender, EventArgs e)
        {
            string fileName = @"..\..\..\..\..\..\Data\DecryptWorkbook.xlsx";

            //Detect if the Excel workbook is password protected.
            bool value = Workbook.IsPasswordProtected(fileName);

            if (value)
            {
                //Load a file with the password specified
                Workbook workbook = new Workbook();
                workbook.OpenPassword = "eiceblue";
                workbook.LoadFromFile(fileName);

                //Decrypt workbook
                workbook.UnProtect();

                //Save the document
                workbook.SaveToFile("DecryptWorkbook_result.xlsx", ExcelVersion.Version2010);

                // Dispose of the workbook object to release resources
                workbook.Dispose();
            }



            ExcelDocViewer("DecryptWorkbook_result.xlsx");
        }
        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}
